clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -846,6 +845,11 @@ private CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> commit(final C
         }
 
         applicationEventHandler.add(commitEvent);
+
+        // Wait for offsets to be ready if none were explicitly specified
+        // This blocks until the background thread retrieves allConsumed positions to commit
```

Comment:
```suggestion
        // This blocks until the background thread retrieves allConsumed positions to commit if none were explicitly specified.
```
(only because both statements seem to say almost the same)